modules/oxide/src/less/theme/components/toolbar-button/toolbar-number-input.less

```diff
@@ -0,0 +1,53 @@
+//
+// Toolbar number input
+//
+
+@toolbar-number-input-button-height: 28px;
+@toolbar-number-input-button-width: 24px;
+@toolbar-number-input-input-wrapper-width: 75px;
+
+.tox {
+  .tox-number-input {
+    display: flex;
+
+    > *:focus > {
+      background: @toolbar-button-focus-background-color;
+      border: @toolbar-button-focus-border;
+      box-shadow: @toolbar-button-focus-box-shadow;
+    }
+
+    .tox-input-wrapper,
+    input {
+      display: flex;
+      text-align: center;
+      width: @toolbar-number-input-input-wrapper-width;
+    }
+
+    button {
+      background: @toolbar-button-bespoke-background-color;
+      border-radius: @toolbar-button-border-radius;
+      height: @toolbar-number-input-button-height;
+      text-align: center;
+      width: @toolbar-number-input-button-width;
+    }
+
+    button:hover {
+      background: @toolbar-button-hover-background-color;
+      border: @toolbar-button-hover-border;
+      box-shadow: @toolbar-button-hover-box-shadow;
+      color: @toolbar-button-hover-text-color;
+
+      svg {
+        fill: @toolbar-button-hover-icon-color;
+      }
+    }
+  }
+
+  .tox-number-input:focus {
+    * > {
```

Comment:
Is this really correct? `* >` isn't valid CSS, so I assume it's a less-ism. I really hope it doesn't apply to `*` because that will match all descendants. But either way, applying these styles to multiple children can't be right:

* background is transparent by default so it will show the parent colour and is best set there
* multiple borders and box shadows on children can't look good

I realise I'm probably missing something, just wanted to make sure it's correct

Reply:
I refactored it, now the only thing that is applied is the `background`, and I also the selector to `.tox-number-input:focus > *` to make clear that the style is applied to all the children.

I did this because the children themselves have a `background` and if I apply it only to the parent nothing happen

Reply:
changed